code/modules/clothing/head/misc_special.dm

```diff
@@ -278,24 +283,42 @@
 		user.gain_trauma(paranoia, TRAUMA_RESILIENCE_MAGIC)
 		to_chat(user, "<span class='warning'>As you don the foiled hat, an entire world of conspiracy theories and seemingly insane ideas suddenly rush into your mind. What you once thought unbelievable suddenly seems.. undeniable. Everything is connected and nothing happens just by accident. You know too much and now they're out to get you. </span>")
 
+
 /obj/item/clothing/head/foilhat/MouseDrop(atom/over_object)
 	//God Im sorry
-	if(usr)
+	if(usr && GetComponent(/datum/component/anti_magic))

```

Comment:
what is this second check for? if you don't have the anti_magic bit you can now take this off freely?

Reply:
Umh, now that you make me think about it, It would be better to assign this check to a var instead.
But yea, basically once the becomes useless from repeated anti_magic stress, the paranoia expires too, and thus the inability to remove it by ourself.